src/main/java/edu/ie3/datamodel/models/input/system/type/chargingpoint/ChargingPointType.java

```diff
@@ -0,0 +1,102 @@
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.models.input.system.type.chargingpoint;
+
+import edu.ie3.datamodel.models.ElectricCurrentType;
+import edu.ie3.util.quantities.dep.PowerSystemUnits;
+import java.util.*;
+import javax.measure.Quantity;
+import javax.measure.quantity.Power;
+
+/**
+ * The actual implementation {@link edu.ie3.datamodel.models.input.system.EvcsInput} types. Default
+ * type implementations as well as methods to parse a type from a string can be found in {@link
+ * ChargingPointTypeUtils}
+ *
+ * @version 0.1
+ * @since 25.07.20
+ */
+public class ChargingPointType {
+
+  private final String id;
+  private final Quantity<Power> sRated;
+  private final ElectricCurrentType electricCurrentType;
+
+  private final Set<String> synonymousIds;
+
+  public ChargingPointType(
+      String id, Quantity<Power> sRated, ElectricCurrentType electricCurrentType) {
+    this.id = id;
+    this.sRated = sRated;
+    this.electricCurrentType = electricCurrentType;
+    this.synonymousIds = new HashSet<>();
+  }
+
+  public ChargingPointType(
+      String id,
+      Quantity<Power> sRated,
+      ElectricCurrentType electricCurrentType,
+      Set<String> synonymousIds) {
+    this.id = id;
+    this.sRated = sRated;
+    this.electricCurrentType = electricCurrentType;
+    this.synonymousIds = synonymousIds;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public Quantity<Power> getsRated() {
+    return sRated;
+  }
+
+  public ElectricCurrentType getElectricCurrentType() {
+    return electricCurrentType;
+  }
+
+  public Set<String> getSynonymousIds() {
+    return synonymousIds;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    ChargingPointType that = (ChargingPointType) o;
+    return id.equals(that.id)
+        && sRated.equals(that.sRated)
+        && electricCurrentType == that.electricCurrentType
+        && synonymousIds.equals(that.synonymousIds);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(id, sRated, electricCurrentType, synonymousIds);
+  }
+
+  @Override
+  public String toString() {
+    return ChargingPointTypeUtils.fromIdString(id)
+        .flatMap(
+            commonType -> {
+              if (commonType.getsRated().equals(sRated)
+                  && commonType.getElectricCurrentType().equals(electricCurrentType)) {
+                return Optional.of(commonType.id);
+              } else {
+                return Optional.empty();
+              }
+            })
+        .orElseGet(
+            () ->
+                id
+                    + "("
+                    + sRated.to(PowerSystemUnits.KILOVOLTAMPERE).getValue().doubleValue()
+                    + "|"
+                    + electricCurrentType
+                    + ")");
+  }
```

Comment:
Could you shed some light on, why this method is that complex? This implies, that there might be a type, that has the same id as a a common type and also the same properties (so a duplicate of the common ones). Then the common one is acquired and it's id is given back. In that case only the id is the result of the `toString`-Method, while in any other case the String looks a bit more detailed?!

Reply:
The idea behind this implementation was, that whenever this type is a common type, the only the easy to read, common id is given back instead of the full id with sRated and electric current type. In contrast, when this is a custom type, we actually need all information, because custom types have custom attributes (sRated, current type).